app/models/field_slip.rb

```diff
@@ -70,7 +70,21 @@ def field_value(field)
   end
 
   def location
-    observation&.place_name || ""
+    result = observation&.place_name || users_last_location
```

Comment:
Instead of `observation&.place_name`, how about `observation&.location` (and below)? That would allow you to also prefill the `location_id` in the autocompleter by sending a `hidden_id:` arg, giving the user that "verified" checkmark, although it may be getting that just fine on this branch.

With the new autocompleters I would expect we are getting actual `Location` records saved on all observations (to be verified...).

Reply:
@JoeCohen here's some context: 

The autocompleter is a lookup device that tries to match a string to a `Location` instance, but in the case of a preselected Location, _**we already have the instance**_. In the template, we can send 
- the `Location.name` to the autocompleter as the text `value`, and 
- the `Location.id` as the `hidden_id`.

The MO app has innumerable duplicate lookups of things by text string. My feeling is that if we are going to improve our provision of geographic UIs, we need to move away from thinking of "locations" as strings we pass around and parse, to passing around MO `Locations`,  instances of geographic data, which we usually already have. 

As it is, MO's code strikes me as a library full of librarians who refuse to use the Dewey decimal system, and only request books from each other by title and author, even when they have the code right in front of them. There's been very little consideration of the cost of these flurries of lookups.

Reply:
I think I see what you're driving at.  However, note that `Location.name` does not do the Postal/Scientific name thing which is why I'm using `place_name`.  There is `Location.display_name` that I could use instead.  Getting the `hidden_id` working would involve creating a FieldSlip.hidden_id, right?  It's not clear to me how that would work since the only use of `hidden_id` I see in the current code base is in `app/javascript/controllers/autocompleter_controller.js`.  What I was hoping you were saying is that if I passed the autocompleter a Location rather than a string that it would take care of the rest.  However, when I tried that I got this:
<img width="623" alt="Screenshot 2024-09-08 at 7 35 01 PM" src="https://github.com/user-attachments/assets/d8a51c5d-e340-4c21-86a3-0b13a2778705">

Reply:
Yes - very close. When building out the autocompleters though, I found it was not always predictable what the identifiers of the two fields should be — like in Observation, for the Location autocompleter, the text field name has to be `place_name` (the visible field of the autocompleter) but i'm expecting the param `location_id` for the hidden field. 

That's why, to keep the autocompleter helper as versatile as needed (not guessing too much about the field names) the two field names and their values can be passed separately (and both are optional). 

To do what we're talking about here, you'd pass:
```ruby
autocompleter_field(form:, field:, value: @location.display_name, 
                    hidden_value: @location.id, hidden_name: #{see note})
```
So: The `hidden_value` saves the AJAX lookup from the text input when the form loads. 

Note: The name of the hidden field `hidden_name` should be whatever `param` you're handling it with in the receiving controller action. In this case, you're not handling it yet, and it can be whatever you like, but fyi the helper defaults to calling it `#{model_type}_id`, e.g., `location_id`. The only controller actions that handle this param currently are the obs create and edit actions, thus saving the second lookup. But i'm looking to roll that out more widely.

Reply:
For reference, here's the relevant part of the HTML the autocompleter generates, the two inputs. The second is the hidden "matched record ids" field:
```html
<input placeholder="Start typing..." autocomplete="off" data-autocompleter-target="input" data-map-target="placeInput" data-action="..." class="form-control" type="text" value="USA, California, Alameda Co., Berkeley" name="observation[place_name]" id="observation_place_name">
<input value="742" data-autocompleter-target="hidden" data-map-target="locationId" data-north="37.9067" data-south="37.8359" data-east="-122.234" data-west="-122.327" class="form-control" readonly="readonly" autocomplete="off" type="hidden" name="observation[location_id]" id="observation_location_id">
```

Reply:
@mo-nathan Updated my (overlong) response for clarity above. 

Sorry for the TMI, i guess it's a case of "in case you're interested this is how it works"